### PR TITLE
Functional Tests : BO - Cart rules - Case 10 - Country Restriction

### PR DESCRIPTION
**Pull request by cfarhani06**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | BO - Cart rules - Case 10 - Country Restriction
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | no
| Related PRs       | no
| How to test?      | no need
| Possible impacts? | no

### CODE REVIEW

**Review by boubkerbribri — DISMISSED** [review]

File: tests/UI/pages/BO/quickAccess/index.js [hunk 1]
```diff
@@ -62,6 +63,15 @@ class QuickAccess extends BOBasePage {
...
 
+  /**
+   * Remove the quick access for current page
+   * @param page {Page} Browser tab
+   * @returns {Promise<void>}
+   */
+  async removeQuickAccessPage(page) {
+    await this.clickAndWaitForNavigation(page, this.removeQuickAccessLink);
+  }
+
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/BO/quickAccess/index.js]: Why did you create this method ? 
I think you didn't use it in your test

File: tests/UI/pages/FO/cart/index.js [hunk 2]
```diff
@@ -164,6 +168,15 @@ class Cart extends FOBasePage {
     return this.getTextContent(page, this.alertWarning);
   }
 
+  /**
+   * Get alert warning
+   * @param page {Page} Browser tab
+   * @returns {Promise<string>}
+   */
+  getAlertWarningForDeliveryAdress(page) {
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/FO/cart/index.js]:
```suggestion
  getAlertWarningForPromoCode(page) {
```

File: tests/UI/pages/BO/quickAccess/index.js (hunk 1 above)
Comment (boubkerbribri) [reply tests/UI/pages/BO/quickAccess/index.js]: I think it's used on the post test [here](https://github.com/PrestaShop/PrestaShop/pull/30049/files#diff-ddf82f3956e39108457032f7d8bc280065a08e1dda61989f64127e0fa7351f2dR97)

Comment (nesrineabdmouleh) [reply tests/UI/pages/BO/quickAccess/index.js]: `await countriesPage.removeLinkFromQuickAccess(page);`
It's not the same method :/

Comment (boubkerbribri) [reply tests/UI/pages/BO/quickAccess/index.js]: you're right, `removeLinkFromQuickAccess` is from the `BoBasePage` so no need for this one probably.

Comment (cfarhani06) [reply tests/UI/pages/BO/quickAccess/index.js]: That's true. I fixed it

File: tests/UI/pages/FO/cart/index.js (hunk 2 above)
Comment (cfarhani06) [reply tests/UI/pages/FO/cart/index.js]: Done

**Review by boubkerbribri — DISMISSED** [review]

File: tests/UI/pages/BO/quickAccess/index.js [hunk 3]
```diff
@@ -19,6 +19,7 @@ class QuickAccess extends BOBasePage {
     // Selectors
     // Header selectors
     this.addNewQuickAccessButton = 'a[data-role=page-header-desc-quick_access-link]';
+    this.removeQuickAccessLink = '#quick-remove-link';
```
Comment (nesrineabdmouleh) [inline tests/UI/pages/BO/quickAccess/index.js]: Please delete the not used selector
Thanks!

Comment (cfarhani06) [reply tests/UI/pages/BO/quickAccess/index.js]: Done

**Review by nesrineabdmouleh — DISMISSED** [review]

File: tests/UI/pages/BO/catalog/discounts/add.js [hunk 4]
```diff
@@ -173,6 +182,20 @@ class AddCartRule extends BOBasePage {
...
+      await page.click(this.firstCountrySelection);
+      await page.click(this.countryGroupRemoveButton);
+    }
+
+    // Set carrier discount
+    if (cartRuleData.carrierRestriction) {
+      await this.setChecked(page, this.carrierRestriction);
+      await page.click(this.carrierRestrictionPickUpInStore);
+      await page.click(this.carrierRestrictionRemoveButton);
+    }
```
Comment (Progi1984) [inline tests/UI/pages/BO/catalog/discounts/add.js]: Duplicate to the line 178.

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/10_countryRestriction.js [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+        this,
+        'testIdentifier',
+        'chooseAndConfirmAdressStep',
+        baseContext,
+      );
+
+      await checkoutPage.chooseDeliveryAddress(page, 2);
+
+      const isDeliveryStep = await checkoutPage.goToDeliveryStep(page);
+      await expect(isDeliveryStep, 'Delivery Step bock is not displayed').to.be.true;
```
Comment (Progi1984) [inline tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/10_countryRestriction.js]:
```suggestion
      await expect(isDeliveryStep, 'Delivery Step block is not displayed').to.be.true;
```

File: tests/UI/pages/BO/quickAccess/index.js [hunk 6]
```diff
@@ -61,7 +61,7 @@ class QuickAccess extends BOBasePage {
   async goToAddNewQuickAccessPage(page) {
     await this.clickAndWaitForNavigation(page, this.addNewQuickAccessButton);
   }
-
+
```
Comment (Progi1984) [inline tests/UI/pages/BO/quickAccess/index.js]:
```suggestion

```

**Review by Progi1984 — COMMENTED** [review]
@cfarhani06 Thank you for your contribution.

3 Feedbacks and it's an :heavy_check_mark: for me.

**Comment by boubkerbribri** [comment]
@cfarhani06 don't forget to run `npm run lint:fix` locally to fix Errors displayed on CI

File: tests/UI/pages/BO/catalog/discounts/add.js (hunk 4 above)
Comment (cfarhani06) [reply tests/UI/pages/BO/catalog/discounts/add.js]: Done

File: tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/10_countryRestriction.js (hunk 5 above)
Comment (cfarhani06) [reply tests/UI/campaigns/functional/BO/03_catalog/07_discounts/01_cartRules/01_CRUDCartRule/10_countryRestriction.js]: Done

File: tests/UI/pages/BO/quickAccess/index.js (hunk 6 above)
Comment (cfarhani06) [reply tests/UI/pages/BO/quickAccess/index.js]: Done

**Review by Progi1984 — APPROVED** [review]

**Review by boubkerbribri — APPROVED** [review]
Thanks @cfarhani06

**Review by jolelievre — APPROVED** [review]

**Pull request merged**